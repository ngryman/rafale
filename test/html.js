import test from 'ava'
import { vnodeToJSON } from './helpers/to-json'
import { html } from '../index'

const noop = () => {}

function macro(t, vtree) {
  t.snapshot(vnodeToJSON(vtree))
}

test('create a simple component', macro,
  html`<b>foo</b>`
)

test('create a component with a primitive child', macro,
  html`<b>${'foo'}</b>`
)

test('create a component with a nested component child', macro,
  html`<p>${
    html`<b>foo</b>`
  }</p>`
)

test('create a component with a sequence child', macro,
  html`<p>${
    ['foo', 'bar'].map(text =>
      html`<b>${text}</b>`
    )
  }</p>`
)

test('create a component with a primitive attribute', macro,
  html`<p id="${'foo'}"></p>`
)

test('create a component with a primitive attribute without quotes', macro,
  html`<p id=${'foo bar'}></p>`
)

test('create a component with mixed attributes', macro,
  html`<p id="${'foo'}" class="bar"></p>`
)

test('create a component with a function attribute', macro,
  html`<button onclick=${noop}></button>`
)

test('create a component with dummy text nodes', macro,
  html`<b>
    ${'foo'}
  </b>`
)

test('remove function attributes but keep others', macro,
  html`<button onclick=${noop} class="foo"></button>`
)
