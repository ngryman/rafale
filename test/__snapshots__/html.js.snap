// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a component with a function attribute 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": [Function],
    },
  ],
  "component": Object {
    "slots": Array [
      Slot {
        "name": "click",
        "path": Array [],
        "type": 1,
      },
    ],
    "template": "<button></button>",
  },
  "type": 2,
  "value": null,
}
`;

exports[`create a component with a nested component child 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "component": Object {
        "slots": Array [],
        "template": "<b>foo</b>",
      },
      "type": 2,
      "value": null,
    },
  ],
  "component": Object {
    "slots": Array [
      Slot {
        "name": null,
        "path": Array [
          0,
        ],
        "type": 0,
      },
    ],
    "template": "<p></p>",
  },
  "type": 2,
  "value": null,
}
`;

exports[`create a component with a primitive attribute 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "foo",
    },
  ],
  "component": Object {
    "slots": Array [
      Slot {
        "name": "id",
        "path": Array [],
        "type": 1,
      },
    ],
    "template": "<p id=\\"ðŸ¦„\\"></p>",
  },
  "type": 2,
  "value": null,
}
`;

exports[`create a component with a primitive attribute without quotes 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "foo bar",
    },
  ],
  "component": Object {
    "slots": Array [
      Slot {
        "name": "id",
        "path": Array [],
        "type": 1,
      },
    ],
    "template": "<p id=\\"ðŸ¦„\\"></p>",
  },
  "type": 2,
  "value": null,
}
`;

exports[`create a component with a primitive child 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "foo",
    },
  ],
  "component": Object {
    "slots": Array [
      Slot {
        "name": null,
        "path": Array [
          0,
        ],
        "type": 0,
      },
    ],
    "template": "<b></b>",
  },
  "type": 2,
  "value": null,
}
`;

exports[`create a component with a sequence child 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "foo",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "component": null,
              "type": 0,
              "value": "bar",
            },
          ],
          "component": Object {
            "slots": Array [
              Slot {
                "name": null,
                "path": Array [
                  0,
                ],
                "type": 0,
              },
            ],
            "template": "<b></b>",
          },
          "type": 2,
          "value": null,
        },
      ],
      "component": Object {
        "slots": Array [
          Slot {
            "name": null,
            "path": Array [
              0,
            ],
            "type": 0,
          },
        ],
        "template": "<b></b>",
      },
      "type": 1,
      "value": null,
    },
  ],
  "component": Object {
    "slots": Array [
      Slot {
        "name": null,
        "path": Array [
          0,
        ],
        "type": 0,
      },
    ],
    "template": "<p></p>",
  },
  "type": 2,
  "value": null,
}
`;

exports[`create a component with dummy text nodes 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "foo",
    },
  ],
  "component": Object {
    "slots": Array [
      Slot {
        "name": null,
        "path": Array [
          0,
        ],
        "type": 0,
      },
    ],
    "template": "<b>
  </b>",
  },
  "type": 2,
  "value": null,
}
`;

exports[`create a component with mixed attributes 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": "foo",
    },
  ],
  "component": Object {
    "slots": Array [
      Slot {
        "name": "id",
        "path": Array [],
        "type": 1,
      },
    ],
    "template": "<p id=\\"ðŸ¦„\\" class=\\"bar\\"></p>",
  },
  "type": 2,
  "value": null,
}
`;

exports[`create a simple component 1`] = `
Object {
  "children": Array [],
  "component": Object {
    "slots": Array [],
    "template": "<b>foo</b>",
  },
  "type": 2,
  "value": null,
}
`;

exports[`remove function attributes but keep others 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "component": null,
      "type": 0,
      "value": [Function],
    },
  ],
  "component": Object {
    "slots": Array [
      Slot {
        "name": "click",
        "path": Array [],
        "type": 1,
      },
    ],
    "template": "<button class=\\"foo\\"></button>",
  },
  "type": 2,
  "value": null,
}
`;
